### PR TITLE
Fix stride printing/parsing formatting

### LINKED ISSUE #41772: [JIT] Change the way strides are printed in types
## 🚀 Feature
The current format for strides in types in JIT IR is the following: `Dtype(size1:stride1, size2:stride2, ..., size_n:stride_n)`, so it looks like, e.g. `Float(30:200, 20:10, 10:1)`. People are often confused by this format, and probably something like `Float(30,20,10, strides=[200,10,1])` would be more user-friendly.

## Test case
```
import torch

def foo():
    current =  '''
        graph(%x : Float(20:10, 10:1)):
            return (%x)
    '''
    graph = torch._C.parse_ir(current)
    print(graph)

    desired =  '''
        graph(%x : Float(20, 10, strides=[10, 1])):
            return (%x)
    '''
```

## Implementation pointers
The change should be made in both IRPrinter and IRParser, since there are tests checking that we can parse what we print. It is expected that such change would break lots of tests, for instance `test/test_jit.py` - we will need to update all the failing tests as well.

#41507 could be used as an example of a similar change.

cc @suo @gmagogsfm

### PR DESCRIPTION
**Pull request by ansley**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45156 Fix stride printing/parsing formatting**

Differential Revision: [D24078695](https://our.internmc.facebook.com/intern/diff/D24078695)

### CODE REVIEW

File: torch/csrc/jit/passes/onnx/preprocess_for_onnx.cpp [hunk 1]
```diff
@@ -37,22 +37,23 @@ at::optional<Node*> FindFusibleListUnpack(Node* n) {
...
+// graph(%input : Float(5, 4, 3, strides=[12, 3, 1])):
 //   %13 : int[] = prim::Constant[value=[2, 1, 2]]()
 //   %7 : int = prim::Constant[value=0]()
 //   %8 : int = prim::Constant[value=3]()  # Adding addtional input of value 3
 //      representing the number of outputs.
-//   %14 : Float(2:12, 4:3, 3:1), %15 : Float(1:12, 4:3, 3:1), %16 : Float(2:12,
-//       4:3, 3:1) = aten::split_with_sizes(%input, %13, %7, %8)
-//   return (%14, %15, %16)
+//   %14 : Float(2, 4, 3, strides=[12, 3, 1]), %15 : Float(1, 4, 3, strides=[12,
+//   3, 1]), %16 : Float(2, 4, 3, strides=[12, 3, 1] =
```
Comment (gmagogsfm) [inline torch/csrc/jit/passes/onnx/preprocess_for_onnx.cpp:55]: Indent line 54 and 55

File: torch/onnx/__init__.py [hunk 2]
```diff
@@ -212,13 +212,13 @@ def export(model, args, f, export_params=True, verbose=False, training=TrainingM
...
             in external binary files and not in the ONNX model file itself. See link for format
-            details: 
+            details:
             https://github.com/onnx/onnx/blob/8b3f7e2e7a0f2aba0e629e23d89f07c7fc0e6a5e/onnx/onnx.proto#L423
             Also, in this case,  argument 'f' must be a string specifying the location of the model.
-            The external binary files will be stored in the same location specified by the model 
+            The external binary files will be stored in the same location specified by the model
             location 'f'. If False, then the model is stored in regular format, i.e. model and
             parameters are all in one file. This argument is ignored for all export types other
-            than ONNX.
```
Comment (gmagogsfm) [inline torch/onnx/__init__.py:221]: It seems this file is still auto-formatted, so you may want to check you editor setting. But I think these space changes are small enough that we can include them in this PR.

File: aten/src/ATen/core/type.cpp [hunk 3]
```diff
@@ -41,10 +41,20 @@ std::ostream& operator<<(std::ostream & out, const Type & t) {
         } else {
           out << "*";
         }
-        if (has_valid_strides_info &&
-            type_verbosity() >= TypeVerbosity::TypeAndStride) {
-          out << ":" << *value->strides()[i];
+      }
+      if (has_valid_strides_info) {
```
Comment (gmagogsfm) [inline aten/src/ATen/core/type.cpp:45]: I might be missing something. Why is the condition here `has_valid_strides_info`? I thought the final `,` be printed as long as `*ndim > 0` regardless of `strides` condition.

File: torch/onnx/__init__.py (hunk 2 above)
Comment (ansleyadelaide) [reply torch/onnx/__init__.py:221]: Good call. I had Editor: Format on Save off already, but I turned Editor > Comments: Insert Space off as well, which I believe should fix the problem going forward.

File: aten/src/ATen/core/type.cpp (hunk 3 above)
Comment (gmagogsfm) [reply aten/src/ATen/core/type.cpp:45]: How about the case where there is complete strides info, but `type_verbosity` is lower than `TypeAndStride`? is `,` needed?

Comment (gmagogsfm) [reply aten/src/ATen/core/type.cpp]: Is this addressed?

**Review by gmagogsfm — APPROVED** [review]

**Comment by ZolotukhinM** [comment]
Wow, nice, thanks for fixing that!

**Pull request closed without merging**